### ISSUE TITLE
Make run target rebuilds and does not work on OSX

### ISSUE THREAD

**Issue by olemis** [fibercrypto/FiberCryptoWallet#257]
**Describe the bug**
`make run` target not working (at least) on osx 

**Environment information:**

- OS: Mac OS X
- Platform: Darwin 17.4.0 x86_64

**Steps to Reproduce**

1. Precondition: Setup project development environment
2. Run `make build`
3. Run `make run`

**Actual behavior**
Executing `make run` in step 3 triggers `make build` once again. On Mac OS nothing gets executed either. 

**Expected behavior**

**Additional context**
There is nothing at path `deploy/linux/FiberCryptoWallet` .

After executing `deploy/darwin/fibercryptowallet.app/Contents/MacOS/fibercryptowallet` from shell `SIGSEGV` is raised .

```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x100000067 pc=0x6533752]

runtime stack:
runtime.throw(0x49f685c, 0x2a)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/panic.go:608 +0x72
runtime.sigpanic()
	/usr/local/Cellar/go/1.11/libexec/src/runtime/signal_unix.go:374 +0x2f2

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x47f5020, 0xc00047baf8, 0x2219d8f0)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/cgocall.go:128 +0x5e fp=0xc00047bac8 sp=0xc00047ba90 pc=0x4007ade
github.com/therecipe/qt/core._Cfunc_QAbstractItemModel_DataChanged(0x17281520, 0x2219b2a0, 0x2219b2a0, 0x2219d8f0)
	_cgo_gotypes.go:528 +0x45 fp=0xc00047baf8 sp=0xc00047bac8 pc=0x4480a35
github.com/therecipe/qt/core.(*QAbstractItemModel).DataChanged.func1(0x17281520, 0x2219b2a0, 0x2219b2a0, 0x2219d8f0)
	$GOPATH/src/github.com/therecipe/qt/core/core-minimal.go:616 +0xf9 fp=0xc00047bb30 sp=0xc00047baf8 pc=0x44ecab9
github.com/therecipe/qt/core.(*QAbstractItemModel).DataChanged(0xc00044e000, 0x4af7cc0, 0xc00059dfb8, 0x4af7cc0, 0xc00059dfb8, 0xc00047bbc0, 0x1, 0x1)
	$GOPATH/src/github.com/therecipe/qt/core/core-minimal.go:616 +0xe3 fp=0xc00047bb78 sp=0xc00047bb30 pc=0x44abde3
github.com/fibercrypto/FiberCryptoWallet/src/models.(*WalletModel).setData(0xc00044e000, 0xc00059dfb8, 0xc0002aef28, 0x104, 0xc00059df01)
	$GOPATH/src/github.com/fibercrypto/FiberCryptoWallet/src/models/walletsModel.go:161 +0x18f fp=0xc00047bbe8 sp=0xc00047bb78 pc=0x455b79f
github.com/fibercrypto/FiberCryptoWallet/src/models.(*WalletModel).editWallet(0xc00044e000, 0xd, 0xc000314280, 0x2, 0x223a2400, 0xc000314285, 0x3, 0xc00031428b, 0x3)
	$GOPATH/src/github.com/fibercrypto/FiberCryptoWallet/src/models/walletsModel.go:197 +0x265 fp=0xc00047bc60 sp=0xc00047bbe8 pc=0x455bdf5
github.com/fibercrypto/FiberCryptoWallet/src/models.(*WalletModel).updateModel(0xc00044e000, 0xc0002720a0, 0x14, 0x14)
	$GOPATH/src/github.com/fibercrypto/FiberCryptoWallet/src/models/walletsModel.go:210 +0x11a fp=0xc00047bcf0 sp=0xc00047bc60 pc=0x455c25a
github.com/fibercrypto/FiberCryptoWallet/src/models.(*WalletModel).updateModel-fm(0xc0002720a0, 0x14, 0x14)
	$GOPATH/src/github.com/fibercrypto/FiberCryptoWallet/src/models/walletsModel.go:67 +0x48 fp=0xc00047bd20 sp=0xc00047bcf0 pc=0x46b9618
github.com/fibercrypto/FiberCryptoWallet/src/models.callbackWalletModel2a8dfa_UpdateModel(0x17281520, 0x2214bfa0, 0x14)
	$GOPATH/src/github.com/fibercrypto/FiberCryptoWallet/src/models/moc.go:3990 +0xab fp=0xc00047bd60 sp=0xc00047bd20 pc=0x45ce9ab
github.com/fibercrypto/FiberCryptoWallet/src/models._cgoexpwrap_d0bd062cf532_callbackWalletModel2a8dfa_UpdateModel(0x17281520, 0x2214bfa0, 0x14)
	_cgo_gotypes.go:25385 +0x3f fp=0xc00047bd88 sp=0xc00047bd60 pc=0x4597c8f
runtime.call32(0x0, 0x7ffeefbfb7f0, 0x7ffeefbfb888, 0x18)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/asm_amd64.s:522 +0x3b fp=0xc00047bdb8 sp=0xc00047bd88 pc=0x405ac1b
runtime.cgocallbackg1(0x0)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/cgocall.go:316 +0x19b fp=0xc00047be30 sp=0xc00047bdb8 pc=0x4007e7b
runtime.cgocallbackg(0x0)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/cgocall.go:194 +0xd6 fp=0xc00047be98 sp=0xc00047be30 pc=0x4007c46
runtime.cgocallback_gofunc(0x4007b02, 0x4828920, 0xc00047bf28, 0x451d37f)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/asm_amd64.s:775 +0x9b fp=0xc00047beb8 sp=0xc00047be98 pc=0x405c1eb
runtime.asmcgocall(0x4828920, 0xc00047bf28)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/asm_amd64.s:622 +0x42 fp=0xc00047bec0 sp=0xc00047beb8 pc=0x405c082
runtime.cgocall(0x4828920, 0xc00047bf28, 0x451a21c)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/cgocall.go:131 +0x82 fp=0xc00047bef8 sp=0xc00047bec0 pc=0x4007b02
github.com/therecipe/qt/gui._Cfunc_QGuiApplication_QGuiApplication_Exec(0x0)
	_cgo_gotypes.go:1357 +0x49 fp=0xc00047bf28 sp=0xc00047bef8 pc=0x471ca59
github.com/therecipe/qt/gui.(*QGuiApplication).Exec(0xc0002dc040, 0x4af7e40)
	$GOPATH/src/github.com/therecipe/qt/gui/gui-minimal.go:1659 +0x22 fp=0xc00047bf40 sp=0xc00047bf28 pc=0x4729502
main.main()
	$GOPATH/src/github.com/fibercrypto/fibercryptowallet/main.go:45 +0x218 fp=0xc00047bf98 sp=0xc00047bf40 pc=0x4744548
runtime.main()
	/usr/local/Cellar/go/1.11/libexec/src/runtime/proc.go:201 +0x207 fp=0xc00047bfe0 sp=0xc00047bf98 pc=0x40312e7
runtime.goexit()
	/usr/local/Cellar/go/1.11/libexec/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc00047bfe8 sp=0xc00047bfe0 pc=0x405c911

goroutine 5 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/sigqueue.go:139 +0x9f
os/signal.loop()
	/usr/local/Cellar/go/1.11/libexec/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
	/usr/local/Cellar/go/1.11/libexec/src/os/signal/signal_unix.go:29 +0x41
```

The error above appears after wallet tree is loaded. At other times the app crashes immediate after `make run` emitting messages shown below

```
$ make run
Running FiberCryptoWallet...
[2019-12-04T17:59:14-05:00] INFO [Connection Pool]: Creating pool section
[2019-12-04T17:59:14-05:00] INFO [Altcoin plugin entry point]: Loading Altcoin manager
[2019-12-04T17:59:14-05:00] INFO [Altcoin plugin entry point]: Register plugin to Altcoin manager
[2019-12-04T17:59:14-05:00] INFO [Altcoin plugin entry point]: Register altcoin to Altcoin manager
[2019-12-04T17:59:14-05:00] INFO [Altcoin plugin entry point]: Register altcoin to Altcoin manager
[2019-12-04T17:59:14-05:00] INFO [Altcoin plugin entry point]: Register altcoin to Altcoin manager
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0xd002fa38 pc=0x6c8bff7]

runtime stack:
runtime.throw(0x4bbe249, 0x2a)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/panic.go:608 +0x72
runtime.sigpanic()
	/usr/local/Cellar/go/1.11/libexec/src/runtime/signal_unix.go:374 +0x2f2

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x4985d00, 0xc0000fbd08, 0xc000000307)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/cgocall.go:128 +0x5e fp=0xc0000fbcd8 sp=0xc0000fbca0 pc=0x4007d0e
github.com/therecipe/qt/qml._Cfunc_QJSEngine_GlobalObject(0x7408330, 0x0)
	_cgo_gotypes.go:258 +0x4a fp=0xc0000fbd08 sp=0xc0000fbcd8 pc=0x45750ea
github.com/therecipe/qt/qml.(*QJSEngine).GlobalObject.func1(0x7408330, 0x4a8f640)
	$GOPATH/src/github.com/therecipe/qt/qml/qml-minimal.go:127 +0x56 fp=0xc0000fbd40 sp=0xc0000fbd08 pc=0x4582396
github.com/therecipe/qt/qml.(*QJSEngine).GlobalObject(0xc0002e6148, 0x4b85645)
	$GOPATH/src/github.com/therecipe/qt/qml/qml-minimal.go:127 +0x5c fp=0xc0000fbd80 sp=0xc0000fbd40 pc=0x45792ec
github.com/therecipe/qt/qml.Z_initEngine(0x4ce4e00, 0xc0002e6148)
	$GOPATH/src/github.com/therecipe/qt/qml/utils-qml.go:30 +0x1e9 fp=0xc0000fbef0 sp=0xc0000fbd80 pc=0x456e6a9
github.com/therecipe/qt/qml.NewQQmlApplicationEngine(0x0, 0x0, 0x0)
	$GOPATH/src/github.com/therecipe/qt/qml/qml-minimal.go:864 +0xeb fp=0xc0000fbf30 sp=0xc0000fbef0 pc=0x457c9bb
main.main()
	$GOPATH/src/github.com/fibercrypto/fibercryptowallet/main.go:42 +0x1d0 fp=0xc0000fbf98 sp=0xc0000fbf30 pc=0x487e040
runtime.main()
	/usr/local/Cellar/go/1.11/libexec/src/runtime/proc.go:201 +0x207 fp=0xc0000fbfe0 sp=0xc0000fbf98 pc=0x4031517
runtime.goexit()
	/usr/local/Cellar/go/1.11/libexec/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc0000fbfe8 sp=0xc0000fbfe0 pc=0x405cb41

goroutine 18 [runnable]:
github.com/therecipe/qt/core._Cfunc_QVariant_DestroyQVariant(0x768e850)
	_cgo_gotypes.go:14997 +0x41
github.com/therecipe/qt/core.(*QVariant).DestroyQVariant.func1(0x768e850)
	$GOPATH/src/github.com/therecipe/qt/core/core-minimal.go:17309 +0x56
github.com/therecipe/qt/core.(*QVariant).DestroyQVariant(0xc000427e10)
	$GOPATH/src/github.com/therecipe/qt/core/core-minimal.go:17309 +0x44

goroutine 19 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/sigqueue.go:139 +0x9f
os/signal.loop()
	/usr/local/Cellar/go/1.11/libexec/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
	/usr/local/Cellar/go/1.11/libexec/src/os/signal/signal_unix.go:29 +0x41
make: *** [run] Error 2
```

**Possible implementation**

- [x] Rebuild binaries only if actually needed
- [x] Fix `make run` so that it execute app on all supported platforms
- ~Document this in README~ ... not necessary
- ~Fix `SIGSEGV` crash noticed shortly after loading app~ ... follow up in #296

**Comment by olemis** [comment]
@olemis issue edited after trying to execute directly from shell session

**Comment by olemis** [comment]
@stdevCrow could you please try to reproduce this on Windows ?

**Comment by olemis** [comment]
I added in issue description another `SIGSEGV` report that must be fixed .

**Comment by olemis** [comment]
only `SEGSIGV` pending after merging #268

**Comment by stdevMac** [comment]
@olemis can you try again if you still get this errors?

**Comment by olemis** [comment]
@stdevMac what brancn ?

**Comment by olemis** [comment]
@stdevMac yes

**Comment by olemis** [comment]
@stdevMac I mean , app still crashing after 2nd or 3rd wallet update

**Comment by olemis** [comment]
@olemis Looks like same error to me . Notice that code changed since issue reported and lines are different , but SIGSEGV panic report indicates it's related to DataChanged

**Comment by olemis** [comment]
@stdevMac this is what `SIGSEGV` error looks like to me now on aba219019d03065d81251e8ea0441d1fc58f46e0

```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0x6ed4752]

runtime stack:
runtime.throw(0x4bb5abd, 0x2a)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/panic.go:608 +0x72
runtime.sigpanic()
	/usr/local/Cellar/go/1.11/libexec/src/runtime/signal_unix.go:374 +0x2f2

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x49273b0, 0xc0005b9ae8, 0x2495d030)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/cgocall.go:128 +0x5e fp=0xc0005b9ab8 sp=0xc0005b9a80 pc=0x400787e
github.com/therecipe/qt/core._Cfunc_QAbstractItemModel_DataChanged(0x19d7db40, 0x2481a970, 0x2481a970, 0x2495d030)
	_cgo_gotypes.go:561 +0x45 fp=0xc0005b9ae8 sp=0xc0005b9ab8 pc=0x444e0a5
github.com/therecipe/qt/core.(*QAbstractItemModel).DataChanged.func1(0x19d7db40, 0x2481a970, 0x2481a970, 0x2495d030)
	$GOPATH/src/github.com/therecipe/qt/core/core-minimal.go:624 +0xf9 fp=0xc0005b9b20 sp=0xc0005b9ae8 pc=0x4511d69
github.com/therecipe/qt/core.(*QAbstractItemModel).DataChanged(0xc0004e3800, 0x4cd7d80, 0xc00047af30, 0x4cd7d80, 0xc00047af30, 0xc0005b9bb0, 0x1, 0x1)
	$GOPATH/src/github.com/therecipe/qt/core/core-minimal.go:624 +0xe3 fp=0xc0005b9b68 sp=0xc0005b9b20 pc=0x44836f3
github.com/fibercrypto/fibercryptowallet/src/models.(*WalletModel).setData(0xc0004e3800, 0xc00047af30, 0xc00047ba30, 0x104, 0xc00047af01)
	$GOPATH/src/github.com/fibercrypto/fibercryptowallet/src/models/walletsModel.go:161 +0x18f fp=0xc0005b9bd8 sp=0xc0005b9b68 pc=0x459bb8f
github.com/fibercrypto/fibercryptowallet/src/models.(*WalletModel).editWallet(0xc0004e3800, 0x11, 0xc0004080e0, 0xc, 0x24554200, 0xc0004080f0, 0x3, 0xc0004080f6, 0x3)
	$GOPATH/src/github.com/fibercrypto/fibercryptowallet/src/models/walletsModel.go:200 +0x265 fp=0xc0005b9c50 sp=0xc0005b9bd8 pc=0x459c225
github.com/fibercrypto/fibercryptowallet/src/models.(*WalletModel).updateModel(0xc0004e3800, 0xc0000be3c0, 0x14, 0x14)
	$GOPATH/src/github.com/fibercrypto/fibercryptowallet/src/models/walletsModel.go:213 +0x11a fp=0xc0005b9ce0 sp=0xc0005b9c50 pc=0x459c68a
github.com/fibercrypto/fibercryptowallet/src/models.(*WalletModel).updateModel-fm(0xc0000be3c0, 0x14, 0x14)
	$GOPATH/src/github.com/fibercrypto/fibercryptowallet/src/models/walletsModel.go:67 +0x48 fp=0xc0005b9d10 sp=0xc0005b9ce0 pc=0x47084b8
github.com/fibercrypto/fibercryptowallet/src/models.callbackWalletModel977bc7_UpdateModel(0x19d7db40, 0x24678c50, 0x14)
	$GOPATH/src/github.com/fibercrypto/fibercryptowallet/src/models/moc.go:25445 +0xab fp=0xc0005b9d50 sp=0xc0005b9d10 pc=0x469560b
github.com/fibercrypto/fibercryptowallet/src/models._cgoexpwrap_c66c0e9e9e91_callbackWalletModel977bc7_UpdateModel(0x19d7db40, 0x24678c50, 0x14)
	_cgo_gotypes.go:33945 +0x3f fp=0xc0005b9d78 sp=0xc0005b9d50 pc=0x45f661f
runtime.call32(0x0, 0x7ffeefbfb6a0, 0x7ffeefbfb738, 0x18)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/asm_amd64.s:522 +0x3b fp=0xc0005b9da8 sp=0xc0005b9d78 pc=0x405a9bb
runtime.cgocallbackg1(0x0)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/cgocall.go:316 +0x19b fp=0xc0005b9e20 sp=0xc0005b9da8 pc=0x4007c1b
runtime.cgocallbackg(0x0)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/cgocall.go:194 +0xd6 fp=0xc0005b9e88 sp=0xc0005b9e20 pc=0x40079e6
runtime.cgocallback_gofunc(0x40078a2, 0x498b3b0, 0xc0005b9f18, 0xc00011fed8)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/asm_amd64.s:775 +0x9b fp=0xc0005b9ea8 sp=0xc0005b9e88 pc=0x405bf8b
runtime.asmcgocall(0x498b3b0, 0xc0005b9f18)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/asm_amd64.s:622 +0x42 fp=0xc0005b9eb0 sp=0xc0005b9ea8 pc=0x405be22
runtime.cgocall(0x498b3b0, 0xc0005b9f18, 0x0)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/cgocall.go:131 +0x82 fp=0xc0005b9ee8 sp=0xc0005b9eb0 pc=0x40078a2
github.com/therecipe/qt/widgets._Cfunc_QApplication_QApplication_Exec(0x0)
	_cgo_gotypes.go:1285 +0x49 fp=0xc0005b9f18 sp=0xc0005b9ee8 pc=0x47f4dc9
github.com/therecipe/qt/widgets.(*QApplication).Exec(0xc0000b2118, 0x4cd7f01)
	$GOPATH/src/github.com/therecipe/qt/widgets/widgets-minimal.go:1242 +0x22 fp=0xc0005b9f30 sp=0xc0005b9f18 pc=0x480fc42
main.main()
	$GOPATH/src/github.com/fibercrypto/fibercryptowallet/main.go:60 +0x2c7 fp=0xc0005b9f98 sp=0xc0005b9f30 pc=0x486cac7
runtime.main()
	/usr/local/Cellar/go/1.11/libexec/src/runtime/proc.go:201 +0x207 fp=0xc0005b9fe0 sp=0xc0005b9f98 pc=0x4031087
runtime.goexit()
	/usr/local/Cellar/go/1.11/libexec/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc0005b9fe8 sp=0xc0005b9fe0 pc=0x405c6b1

goroutine 20 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/Cellar/go/1.11/libexec/src/runtime/sigqueue.go:139 +0x9f
os/signal.loop()
	/usr/local/Cellar/go/1.11/libexec/src/os/signal/signal_unix.go:23 +0x22
created by os/signal.init.0
	/usr/local/Cellar/go/1.11/libexec/src/os/signal/signal_unix.go:29 +0x41
make: *** [run] Error 2

```

**Comment by olemis** [comment]
Work to fix `SIGSEGV` crash will be continued in #296
